compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -2328,6 +2328,7 @@ fn add_upstream_native_libraries(
         .find(|(ty, _)| *ty == crate_type)
         .expect("failed to find crate type in dependency format list");
 
+    let search_path = archive_search_paths(sess);
```

Comment:
Looks like a good candidate for `std::lazy::Lazy`, it should be evaluated at most once in the loop below, but it would be better to not evaluate it if it's not actually used.

Reply:
Good idea!